quantity_warn_on_lot/models/stock_move_split.py

```diff
@@ -0,0 +1,185 @@
+        view_ref = ir_model_data.get_object_reference(
+            cr, uid, 'quantity_warn_on_lot', 'view_confirm_split'
+        )
+        return view_ref and view_ref[1] or False
+
+    def restore_split(self, cr, uid, ids, context=None):
+        """Restore the split to where we were before the warning."""
+        res = {
+            'type': 'ir.actions.act_window',
+            'res_model': 'stock.move.split',
+            'res_id': ids[0],
+            'view_type': 'form',
+            'view_mode': 'form',
+            'target': 'new',
+            'nodestroy': True,
+            'context': context,
+        }
+
+        active_id = context.get('active_id')
+        move = self.pool['stock.move'].browse(cr, uid, active_id)
+        picking = move.picking_id
+
+        picking.message_post(
+            _("Pressed correct button in the confirmation window")
+        )
+
+        return res
+
+    def continue_split(self, cr, uid, ids, context=None):
+        """Continue splitting without and silence the warning."""
+        if context is None:
+            context = {}
+
+        # Adding continue_split to the context will silence
+        # the check for amount_quantity
+        context['continue_split'] = True
+
+        active_id = context.get('active_id')
+        move = self.pool['stock.move'].browse(cr, uid, active_id)
+        picking = move.picking_id
+
+        picking.message_post(
+            _("Pressed continue button in the confirmation window")
+        )
+
+        return self.split_lot(cr, uid, ids, context=context)
+
+    def fields_view_get(
+        self, cr, uid,
+        view_id=None, view_type='form',
+        context=None, toolbar=False,
+        submenu=False
+    ):
+        """Override the confirm_view and add a custom message."""
+        base_func = super(StockMoveSplit, self).fields_view_get
+        ret = base_func(cr, uid, view_id, view_type, context=context)
+
+        view_confirm_id = self.get_confirm_view_id(cr, uid, context)
+
+        if view_confirm_id == view_id:
+            # Edit only the confirm window
+            message = context.get('warning_message', '')
+            # Simple and dirty replace {message} by our text
+            ret['arch'] = ret['arch'].replace('{message}', message)
+
+        return ret
+
+    def split_lot(self, cr, uid, ids, context=None):
+        """
+        Override the split_lot method and handle errors.
+
+        In case the available quantity is below 0, the split
+        method will raise an InvalidAmountException. This method
+        catch this error and instead open a new view that requires
+        confirmation to continue the split or keep editing.
+
+        Before returning this method, we have to rollback changes to
+        be certain that none of the change made before raising the
+        error will get saved to the database.
+        """
+        base_func = super(StockMoveSplit, self).split_lot
+
+        try:
```

Comment:
@joaoalf I understand. The goal of the module is to add a check before the split. Which means that the user has to click on the split and if something wrong happens, it will show the popup. Otherwise it does as it always did. Though, as you say, it seems a bit strange and the wizard part isn't very elegant though. The other problem is that it is tracking which button are pressed.

A validation button wouldn't speak as much as a dialog box saying "You're doing something wrong. But you can keep going with it or correct it." That said, the check is just a warning and doesn't override the real validation that the split button does.